### ISSUE TITLE
Output export problems

### ISSUE THREAD

**Issue by idealkindom** [Proektsoftbg/Calcpad#212]
1. When export the output to a docx file, the images displayed nomarlly in the output window got shortened,just liked been pressed downwards from the top;

2. When export the output to a pdf file, the images become Ok, but the font of subscripts of symbols become very wierd, i.e., not in harmony with the symbol's main body.

**Comment by Proektsoftbg** [comment]
Hi! 

Can you please post an example - sample code and screenshot.
Try to set both dimensions of the picture - width and height, e.g.: 
`<img style="width: 400pt; height:300pt;" src="..." alt="...">`

**Comment by idealkindom** [comment]
> Hi!
> 
> Can you please post an example - sample code and screenshot. Try to set both dimensions of the picture - width and height, e.g.: `<img style="width: 400pt; height:300pt;" src="..." alt="...">`

Sorry, I left the code in the PC of my office. However, I am sure that in order to align 3 images in one row orderly, I only define the height as“ heigh:50mm;”. Another reason is that  I don't want to calculate the dimension to keep the width to height ratio to make the image look good.

The fonts in docx and pdf can be testified easy since you hava accumulated a lot of codes.

**Comment by Proektsoftbg** [comment]
Hi, again!

Since there is no 1 to 1 compatibility between Html and docx, not everything from Html can work in the same way. I faced the same problem and found that you have to specify both dimensions. I can think about some workaround in the future, but for now, this is the solution.

About the fonts, the problem is that it looks fine on my side:
![image](https://github.com/Proektsoftbg/Calcpad/assets/90918363/8adbe584-99a9-46be-9701-335d0eba2e19)

Maybe you are missing some fonts? Check if you have Calibri and try to reinstall it. Try to open the pdf with another viewer, MS Edge works fine for me, but try a few to see if it will make any difference.

**Comment by idealkindom** [comment]
> Hi, again!
> 
> Since there is no 1 to 1 compatibility between Html and docx, not everything from Html can work in the same way. I faced the same problem and found that you have to specify both dimensions. I can think about some workaround in the future, but for now, this is the solution.
> 
> About the fonts, the problem is that it looks fine on my side:
> 
> ![image](https://user-images.githubusercontent.com/90918363/273391834-8adbe584-99a9-46be-9701-335d0eba2e19.png)
> 
> Maybe you are missing some fonts? Check if you have Calibri and try to reinstall it. Try to open the pdf with another viewer, MS Edge works fine for me, but try a few to see if it will make any difference.

So to define both height and width is a must with html, then I will check it.
I see "season" as subscription, and I don't think it's the same font with the "C". On one PC, I understand it should not attribute to font lack.

Anyway, I will check it again on my working PC in office.

One more thing, I like the way you defined the flip_trigger in the example of "SLS Design of Beam with Rectangular Section". I could understand the code fully and well, I learned it from "copy and revision", however, if I want to do a file with multiple cases of image flip trigger, I found it's not  so easy.

I hope this fancy operation could be added into the submenu of Html UI.Is that possible?

**Comment by Proektsoftbg** [comment]
No, subscripts are not the same font as variables, but it is the same in the html. This is not a bug, it is made like that on purpose, by design. Subscript letters are smaller and they look better in sans-serif font, while variables are in a serif font. In Word, fonts for the equations are not set. Word works in a different way. All equations in Word use the default font, which is Cambria Math. As I mentioned, not all Html formatting can be converted one-to-one in Word. 

BTW, it you want to change the font for the subscript, you can do int in Calcpad with CSS. Try to place the following line on top of your worksheet:
`'<style>sub {font-family:"Times New Romans"!Important;}</style> `

If you want to change it permanently, find ".eq sub " in the template (c:\Program Files\Calcpad\template.html) and replace the font there.

**Comment by Proektsoftbg** [comment]
Yes, flip_trigger is a nice feature. I will consider embedding it into the application. About multiple flip triggers, one colleague from Italy did such thing. I will find it and post an example here.

**Comment by idealkindom** [comment]
Ned.I will try to define the fonts.Another news is that I tried to align images in table and the output to docx is quite good.Just another way.

在 2023-10-07 23:19:22，"Ned Ganchovski" ***@***.***> 写道：


No, subscripts are not the same font as variables, but it is the same in the html. This is not a bug, it is made like that on purpose, by design. Subscript letters are smaller and they look better in sans-serif font, while variables are in a serif font. In Word, fonts for the equations are not set. Word works in a different way. All equations in Word use the default font, which is Cambria Math. As I mentioned, not all Html formatting can be converted one-to-one in Word.

BTW, it you want to change the font for the subscript, you can do int in Calcpad with CSS. Try to place the following line on top of your worksheet:
'<style>sub {font-family:"Times New Romans"!Important;}</style>

If you want to change it permanently, find ".eq sub " in the template (c:\Program Files\Calcpad\template.html) and replace the font there.

**Comment by idealkindom** [comment]
Ned,it looks fancy but fragile.I found if you use some #input before the flip trigger selection in one worksheet, it always cause the app to get stuck.Even sometimes you make it run, the flip trigger selection can not work anu more.

Another problem is that since cpdz could be included in a cpd, then the code of cpdz in fact become breached, right? The logic shoud be optimized.

Another unimportant thing is when you click the open menu and pull selct file type to be txt, then browse a drive route. It's supprising that it could not find any txt files in that location.

在 2023-10-08 18:09:25，"Ned Ganchovski" ***@***.***> 写道：


Yes, flip_trigger is a nice feature. I will consider embedding it into the application. About multiple flip triggers, one colleague from Italy did such thing. I will find it and post an example here.

**Comment by idealkindom** [comment]
> Yes, flip_trigger is a nice feature. I will consider embedding it into the application. About multiple flip triggers, one colleague from Italy did such thing. I will find it and post an example here.

I made it to align images in table and the export to docx file is quite good. However, it incurrs the side effect, i.e., the export to pdf with the same output is abnormal for all the images in tables disappeared.

**Comment by Proektsoftbg** [comment]
> Ned,it looks fancy but fragile.I found if you use some #input before the flip trigger selection in one worksheet, it always cause the app to get stuck.Even sometimes you make it run, the flip trigger selection can not work anu more. Another problem is that since cpdz could be included in a cpd, then the code of cpdz in fact become breached, right? The logic shoud be optimized. Another unimportant thing is when you click the open menu and pull selct file type to be txt, then browse a drive route. It's supprising that it could not find any txt files in that location.

OK. Thank you for your feedback. I will take a closer look. 

I would highly appreciate if you also provide sample code or screenshots, when reporting a problem. You have it anyway. Otherwise, it sometimes takes a lot of time to figure out what exactly is the problem and to reproduce it. It is also possible to make it in a different way that works and not reproduce it at all. Especially when Html is involved, it may be just some incomplete Html code that works in the browser, but cannot convert.

**Comment by idealkindom** [comment]
> > Ned,it looks fancy but fragile.I found if you use some #input before the flip trigger selection in one worksheet, it always cause the app to get stuck.Even sometimes you make it run, the flip trigger selection can not work anu more. Another problem is that since cpdz could be included in a cpd, then the code of cpdz in fact become breached, right? The logic shoud be optimized. Another unimportant thing is when you click the open menu and pull selct file type to be txt, then browse a drive route. It's supprising that it could not find any txt files in that location.
> 
> OK. Thank you for your feedback. I will take a closer look.
> 
> I would highly appreciate if you also provide sample code or screenshots, when reporting a problem. You have it anyway. Otherwise, it sometimes takes a lot of time to figure out what exactly is the problem and to reproduce it. It is also possible to make it in a different way that works and not reproduce it at all. Especially when Html is involved, it may be just some incomplete Html code that works in the browser, but cannot convert.

OK.I will try to feed back in this way.

**Comment by berkdemir** [comment]
> Try to set both dimensions of the picture - width and height, e.g.:
> <img style="width: 400pt; height:300pt;" src="..." alt="...">

I tend to use only the width (because I don't want to calculate the aspect ratio and give the new height), and I have seen this also in the word export. I manually corrected it.
